kube-controllers/pkg/controllers/loadbalancer/loadbalancer_controller.go

```diff
@@ -296,56 +295,69 @@ func (c *loadBalancerController) handleUpdate(update interface{}) {
 			}
 		case model.BlockKey:
 			c.handleBlockUpdate(update)
+			kick(c.syncChan)
 			return
 		}
 	}
 }
 
 func (c *loadBalancerController) handleBlockUpdate(kvp model.KVPair) {
```

Comment:
We should make sure to write good UTs for this as well (i.e., sending a series of block updates and checking that the internal allocationTracker maps are updated correctly after each).